gateway/ln-gateway/src/gateway_module_v2/send_sm.rs

```diff
@@ -234,4 +255,32 @@ impl SendStateMachine {
             Err(e) => old_state.update(SendSMState::Cancelled(e)),
         }
     }
+
+    fn outgoing_payment_success_event(
+        common: SendSMCommon,
+        is_swap: bool,
+    ) -> OutgoingPaymentSuccess {
+        let invoice_amount = match common.invoice {
+            LightningInvoice::Bolt11(invoice) => Amount::from_msats(
+                invoice
+                    .amount_milli_satoshis()
+                    .expect("amountless invoices are not supported"),
+            ),
+        };
+
+        let end_millis = now()
```

Comment:
Yes, since you're capturing `end_millis` at the moment the event is being added, it will virtually equal to the automatically generated timestamp value.

Reply:
Removed this. We can rely on the timestamp of the event